src/jtag3.c

```diff
@@ -2402,6 +2402,49 @@ static int jtag3_set_sck_period(const PROGRAMMER *pgm, double v) {
 }
 
 
+static int jtag3_get_sck_period(const PROGRAMMER *pgm, double *v) {
+  unsigned char conn, arch;
+  unsigned char buf[2];
+  *v = 0;
+
+  if (jtag3_getparm(pgm, SCOPE_AVR, 1, PARM3_CONNECTION, &conn, 1) < 0) {
+    pmsg_error("%s(): cannot obtain connection type\n", __func__);
```

Comment:
Pro tip: the messaging system for OS errors, errors and warnings automatically displays the function name on `-v` [and  file name and line number on `-vvv`]. So, no need for `"%s(): ...", __func__` in the message.

Reply:
Thanks! I'll continue working on this PR tonight